lib/truffle/io/console.rb

```diff
@@ -45,6 +45,38 @@ def getpass(prompt = nil)
 
 
 class IO
+  def self.console(sym=nil, *args)
+    raise TypeError, "expected Symbol, got #{sym.class}" unless sym.nil? || sym.kind_of?(Symbol)
+
+    # klass = self == IO ? File : self
+    if defined?(@console) # using ivar instead of hidden const as in MRI
+      con = @console
+      # MRI checks IO internals : (!RB_TYPE_P(con, T_FILE) || (!(fptr = RFILE(con)->fptr) || GetReadFD(fptr) == -1))
+      if !con.kind_of?(File) || (con.kind_of?(IO) && (con.closed? || !FileTest.readable?(con)))
```

Comment:
This is weird, `con.kind_of?(IO)` can only be true and could be removed, no?

Reply:
Also, since `@console` is only set below to a File, the first check could be simplified to just `if con` (not nil).

Reply:
It's a defensive check ported from MRI. I think the idea is to prevent people from mucking with the internal state.

Reply:
Seems fine, but I'd remove the `con.kind_of?(IO)` since that's just redundant and doesn't seem to be in MRI's code.